File: jorgecunha07-mgt-0035386b5c0c/jorgecunha07-mgt-0035386b5c0c/Mappers/TaskMapper.cs
using MGT.DTO;
using MGT.Entities;
using MGT.Enums;

namespace MGT.Mappers;

public static class TaskMapper
{
    public static TaskDto ToDto(Task task)
    {
        return new TaskDto
        {
            TaskStatus = task.TaskStatus,
            Description = task.Description ?? string.Empty,
            User = task.User,
            RobotId = task.RobotId ?? string.Empty,
            RobotType = task.RobotType ?? string.Empty,
            Name = task.Name,
            TaskType = task.TaskType,
            FromLocation = LocationMapper.ToDto(task.FromLocation),
            ToLocation = LocationMapper.ToDto(task.ToLocation)
        };
    }

    public static Task ToEntity(TaskDto taskDto)
    {
        return new Task
        {
            TaskStatus = TaskStatusEnum.Submitted,
            Description = taskDto.Description,
            User = taskDto.User,
            RobotId = taskDto.RobotId,
            RobotType = taskDto.RobotType,
            TaskType = taskDto.TaskType,
            Name = taskDto.Name,
            FromLocation = LocationMapper.ToEntity(taskDto.FromLocation),
            ToLocation = LocationMapper.ToEntity(taskDto.ToLocation)
        };
    }
                                                                                    
}

File: jorgecunha07-frontend-1dcc492a1c33/jorgecunha07-frontend-1dcc492a1c33/src/app/Mappers/TaskMapper.ts
import { TaskStatusEnum } from "../Constants/TastStatusEnum";
import { TaskTypeEnum } from "../Constants/TaskTypeEnum";
import {Task_Model} from "../models/Task_Model";
import {TaskDTO} from "../dto/TaskDTO";

// Assuming Task_Model and TaskDTO are already imported or defined here

export class TaskMapper {
  public static toTaskDTO(taskModel: Task_Model): TaskDTO {
    return {
      id: taskModel.id,
      taskStatus: taskModel.taskStatus,
      description: taskModel.description,
      taskType: taskModel.taskType,
      user: taskModel.user,
      robotId: taskModel.robotId,
      robotType: taskModel.robotType,
      name: taskModel.name,
    };
  }

  public static toTaskModel(taskDTO: TaskDTO): Task_Model {
    return {
      id: taskDTO.id,
      taskStatus: taskDTO.taskStatus,
      description: taskDTO.description,
      taskType: taskDTO.taskType,
      user: taskDTO.user,
      robotId: taskDTO.robotId,
      robotType: taskDTO.robotType,
      name: taskDTO.name,
    };
  }
}
